### PR TITLE
[4.2] Use the model for edit check

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
The access level edit check for super admins should use the model to fetch the level.

### Testing Instructions
- Create a user which can log in on the back end with the following permissions for the users component  
image
- Log in as that user on the back end
- Open the access level list view in the back end
- Create a new access level without no group assigned
- Save and close that level
- Click on the new access level in the list which should redirect then to the access level form

### Actual result BEFORE applying this Pull Request
All works but a warning is logged:
_Warning: Undefined array key 0 in /administrator/components/com_users/src/Controller/LevelController.php on line 70_

### Expected result AFTER applying this Pull Request
All works and no warning is logged.

### CODE REVIEW

File: administrator/components/com_users/src/Controller/LevelController.php [hunk 1]
```diff
@@ -64,22 +64,10 @@ protected function allowSave($data, $key = 'id')
 	protected function allowEdit($data = array(), $key = 'id')
 	{
 		// Check for if Super Admin can edit
-		$data['id'] = (int) $data['id'];
-		$db    = Factory::getDbo();
```
Comment (Quy) [inline administrator/components/com_users/src/Controller/LevelController.php:68]: Please remove imports.

**Comment by toivo** [comment]
I have tested this item :red_circle: unsuccessfully on 697784802cc4dafcd87d4f90b1bdbad33ff91a6a<br /><br />Tested in Joomla 4.2.9-alpha2-dev of 25 April in Wampserver 3.2.8 using PHP 8.0.15 and MySQL 8.0.27.
The following warning was displayed before and after the patch was applied in Users: Viewing Access Levels when logged in as the test user with the specified Manager access:

[26-Apr-2022 01:36:15 UTC] PHP Warning:  Undefined array key 0 in C:\www\joomla42\administrator\components\com_users\tmpl\levels\default.php on line 86
[26-Apr-2022 01:36:15 UTC] PHP Stack trace:
[26-Apr-2022 01:36:15 UTC] PHP   1. {main}() C:\www\joomla42\administrator\index.php:0
[26-Apr-2022 01:36:15 UTC] PHP   2. require_once() C:\www\joomla42\administrator\index.php:32
[26-Apr-2022 01:36:15 UTC] PHP   3. Joomla\CMS\Application\CMSApplication->execute() C:\www\joomla42\administrator\includes\app.php:63
[26-Apr-2022 01:36:15 UTC] PHP   4. Joomla\CMS\Application\AdministratorApplication->doExecute() C:\www\joomla42\libraries\src\Application\CMSApplication.php:278
[26-Apr-2022 01:36:15 UTC] PHP   5. Joomla\CMS\Application\AdministratorApplication->dispatch($component = *uninitialized*) C:\www\joomla42\libraries\src\Application\AdministratorApplication.php:186
[26-Apr-2022 01:36:15 UTC] PHP   6. Joomla\CMS\Component\ComponentHelper::renderComponent($option = 'com_users', $params = *uninitialized*) C:\www\joomla42\libraries\src\Application\AdministratorApplication.php:143
[26-Apr-2022 01:36:15 UTC] PHP   7. Joomla\CMS\Dispatcher\ComponentDispatcher->dispatch() C:\www\joomla42\libraries\src\Component\ComponentHelper.php:389
[26-Apr-2022 01:36:15 UTC] PHP   8. Joomla\CMS\MVC\Controller\BaseController->execute($task = 'display') C:\www\joomla42\libraries\src\Dispatcher\ComponentDispatcher.php:146
[26-Apr-2022 01:36:15 UTC] PHP   9. Joomla\Component\Users\Administrator\Controller\DisplayController->display($cachable = *uninitialized*, $urlparams = *uninitialized*) C:\www\joomla42\libraries\src\MVC\Controller\BaseController.php:735
[26-Apr-2022 01:36:15 UTC] PHP  10. Joomla\CMS\MVC\Controller\BaseController->display($cachable = FALSE, $urlparams = []) C:\www\joomla42\administrator\components\com_users\src\Controller\DisplayController.php:135
[26-Apr-2022 01:36:15 UTC] PHP  11. Joomla\Component\Users\Administrator\View\Levels\HtmlView->display($tpl = *uninitialized*) C:\www\joomla42\libraries\src\MVC\Controller\BaseController.php:697
[26-Apr-2022 01:36:15 UTC] PHP  12. Joomla\CMS\MVC\View\HtmlView->display($tpl = NULL) C:\www\joomla42\administrator\components\com_users\src\View\Levels\HtmlView.php:91
[26-Apr-2022 01:36:15 UTC] PHP  13. Joomla\CMS\MVC\View\HtmlView->loadTemplate($tpl = NULL) C:\www\joomla42\libraries\src\MVC\View\HtmlView.php:220
[26-Apr-2022 01:36:15 UTC] PHP  14. include() C:\www\joomla42\libraries\src\MVC\View\HtmlView.php:436
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37637">issues.joomla.org/tracker/joomla-cms/37637</a>.</sub>

**Comment by laoneo** [comment]
This issue is not related to the pr, but I fixed it here as well. Thanks for the report.

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 50da8d160daaf7c7742dcd6c82525fd238a6b722<br /><br />Tested successfully in Joomla 4.2.0-alpha3-dev of 3 May in Wampserver 3.2.8 using PHP 8.0.15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37637">issues.joomla.org/tracker/joomla-cms/37637</a>.</sub>

**Comment by pabloarias** [comment]
I have tested this item :white_check_mark: successfully on 50da8d160daaf7c7742dcd6c82525fd238a6b722<br /><br />Tested with Joomla 4.2-alpha3-dev and PHP 8.1.5.

Warning logged before apply the path is a little bit different:

PHP Warning: Undefined array key 0 in /administrator/components/com_users/src/Controller/LevelController.php on line 82<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37637">issues.joomla.org/tracker/joomla-cms/37637</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37637">issues.joomla.org/tracker/joomla-cms/37637</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**